egret/parsers/prescient_dat_parser.py

```diff
@@ -379,11 +382,9 @@ def nd_gen_init(m,b):
     model.NondispatchableGeneratorsAtBus = Set(model.Buses, initialize=nd_gen_init)
     
     def NonNoBus_init(m):
-        retval = set()
         for b in m.Buses:
-            retval = retval.union([gen for gen in m.NondispatchableGeneratorsAtBus[b]])
```

Comment:
Ugh - this was truly ugly and about as maximally inefficient as one could get!

Reply:
Yeah -- the code replacing this isn't identical in functionality. But two generators with the same name at different buses should be caught anyways.